modules/apps/headless-apio/forms/forms-apio-impl/src/main/java/com/liferay/forms/apio/internal/architect/resource/FormInstanceNestedCollectionResource.java

```diff
@@ -98,14 +101,17 @@ public String getName() {
 			new EvaluateContextRoute(), this::_evaluateContext,
 			AcceptLanguage.class, DDMFormRenderingContext.class,
 			ThemeDisplay.class, FormContextIdentifier.class,
-			this::_hasPermission, FormContextForm::buildForm
+			failOnException(_hasPermission.forAddingIn(
+				FormInstanceRecordIdentifier.class)), FormContextForm::buildForm
 		).addCustomRoute(
 			new FetchLatestDraftRoute(), this::_fetchDDMFormInstanceRecord,
 			CurrentUser.class, FormInstanceRecordIdentifier.class,
-			this::_hasPermission, FetchLatestDraftForm::buildForm
+			failOnException(_hasPermission.forAddingIn(
```

Comment:
It's the same method call, right?

Reply:
yes, it's the same but, using the permission util function `failOnException` will return false if there is an exception